### PR TITLE
removed vscode compilation & use vscode-web package

### PR DESCRIPTION
**Pull request by Felx-B**
Hi there,
following my comment in #62 I tried to remove the vscode compilation from the project.
Here is my first attempt if you are still interested.

But some features are still missing 
* Home button / icon is no more available
* read-only tip is displayed on all file tabs
* Welcome page is no more customizable
* Notification warning banner has to be done externally
* Loader

Let me know what you think !

### CODE REVIEW

File: resources/index.html [hunk 1]
```diff
@@ -109,50 +116,82 @@
...
+
 	<body aria-label="">
 		<noscript title="No JavaScript Support">
 			<h1>You need to enable JavaScript to run this app.</h1>
 		</noscript>
-		<div id='load-spinner' aria-label="loading">
+		<!-- <div id='load-spinner' aria-label="loading">
 			<div class="lds-roller"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
-		</div>
+		</div> -->
```
Comment (xcv58) [inline resources/index.html:126]: Loader is disabled by this comment.

**Comment by xcv58** [comment]
It looks great. But the `yarn watch` command is failing, and when I comment this line https://github.com/Felx-B/github1s/blob/master/scripts/watch.sh#L40, the `yarn watch` finish immediately.

For these points:
- Home button / icon is no more available
- read-only tip is displayed on all file tabs

I think we could use extension to achieve the same https://code.visualstudio.com/api/extension-capabilities/extending-workbench

- Welcome page is no more customizable
Use the webview api to render the page https://code.visualstudio.com/api/extension-guides/webview 

- Notification warning banner has to be done externally
maybe use vanilla js + cookie to put a fixed banner

- Loader
https://github.com/conwnet/github1s/pull/118/files#r575888491

File: resources/index.html (hunk 1 above)
Comment (xcv58) [reply resources/index.html:126]: I see, the issue is how we delete the element after the vscode-web loaded in browser.

This patch will work:


```
diff --git a/resources/index.html b/resources/index.html
index 9f05f8b..c014364 100644
--- a/resources/index.html
+++ b/resources/index.html
@@ -121,12 +121,42 @@
 		<noscript title="No JavaScript Support">
 			<h1>You need to enable JavaScript to run this app.</h1>
 		</noscript>
-		<!-- <div id='load-spinner' aria-label="loading">
+		<div id='load-spinner' aria-label="loading">
 			<div class="lds-roller"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
-		</div> -->
+		</div>
 	</body>
 
 	<script>
+		function waitForElement(selector) {
+			return new Promise(function(resolve, reject) {
+				const element = document.querySelector(selector);
+
+				if(element) {
+				resolve(element);
+				return;
+				}
+
+				const observer = new MutationObserver(function(mutations) {
+				mutations.forEach(function(mutation) {
+					const nodes = Array.from(mutation.addedNodes);
+					for(const node of nodes) {
+					if(node.matches && node.matches(selector)) {
+						observer.disconnect();
+						resolve(node);
+						return;
+					}
+					};
+				});
+				});
+
+				observer.observe(document.documentElement, { childList: true, subtree: true });
+			});
+		}
+		waitForElement('div[role="application"]').then(function() {
+			document.querySelector('#load-spinner').remove();
+		});
+
+
 
 parseGitHubUrl = (url) => {
 			const urlObj = new window.URL(url);
```

**Comment by Felx-B** [comment]
I think I can easily manage notification banner & loader.
Welcome page is not yet customizable, but some work are planned for this iteration https://github.com/microsoft/vscode/issues/116000

But home button & read-only tips I won't be able to manage it through extension points.

@conwnet What kind of stuff, do you think, we need to keep ? 
Home button & read only tips seem not essential. And welcome page is almost not visible as we open Readme file by default.

All info in Welcome page can be moved to Github1s:settings panel.
Link to github repository can be moved to status bar.

**Comment by conwnet** [comment]
> I think I can easily manage notification banner & loader.
> Welcome page is not yet customizable, but some work are planned for this iteration [microsoft/vscode#116000](https://github.com/microsoft/vscode/issues/116000)
> 
> But home button & read-only tips I won't be able to manage it through extension points.
> 
> @conwnet What kind of stuff, do you think, we need to keep ?
> Home button & read only tips seem not essential. And welcome page is almost not visible as we open Readme file by default.
> 
> All info in Welcome page can be moved to Github1s:settings panel.
> Link to github repository can be moved to status bar.

In my opinion, we should change the source code of vscode as little as we can, but I think there are somethings we should care.

If we can't change the source code of vscode,  when we want do a little change for better user experience, we have to wait the vscode officially (or never achieve it), you can search `NOTE@coder` in the [code-server](https://github.com/cdr/code-server), you will fount many reasons to the change the source code of vscode to build a browser version.

On the other hand, GitHub1s just want to provide a better experience for view code, we may remove some unused code in vscode for a minimal bundle in the furture,

The vscode-web is great, I think it can provide an build script for better development experience.

**Comment by Felx-B** [comment]
Ok I understand, I close this PR

**Comment by xcv58** [comment]
> Ok I understand, I close this PR

I did try your PR and really enjoy it because it's so fast to build and iterate.

Maybe we could do the following to finally integrate with vscode-web:

1. Identity the places we need to change the source code of VS Code
2. Put those features into two groups: a. doable via extension API, b. impossible via extension API
3. Create issues for `doable via extension API`
4. Identify how could we achieve `impossible via extension API`. I have some ideas like, add hook APIs in vscode-web, use vanilla JS to dynamically inject UI view

**Pull request closed without merging**